dokka-subprojects/plugin-base/src/main/kotlin/org/jetbrains/dokka/base/resolvers/shared/PackageList.kt

```diff
@@ -14,8 +14,7 @@ public data class PackageList(
     val locations: Map<String, String>,
     val url: URL
 ) {
-    val packages: Set<String>
-        get() = modules.values.flatten().toSet()
+    val packages: Set<String> by lazy { modules.values.flatten().toSet() }
 
     public fun moduleFor(packageName: String): Module? {
         return modules.asSequence()
```

Comment:
Also, it seems this could be rewritten a bit better:
e.g. `modules.entries.firstOrNull { it.value.contains(packageName) }?.key`

Reply:
Thanks!
Keeping in mind this function (which I initially just ignored as there were no hot-spot in it), I've rewritten implementation so that `moduleFor` will be just map entry access.
I've also rerun perf tests on the same projects and the result is roughly the same.